Workaround for: https://bugs.launchpad.net/percona-server/+bug/747152

When linking plugins which are linked to libraries already linked with
probes_mysql.o, skip adding (second) probes_mysql.o .

--- Percona-Server-5.5.32-rel31.0/cmake/plugin.cmake.orig	2013-07-01 05:16:34.000000000 +0200
+++ Percona-Server-5.5.32-rel31.0/cmake/plugin.cmake	2013-07-12 17:57:12.120001411 +0200
@@ -47,7 +47,7 @@
 MACRO(MYSQL_ADD_PLUGIN)
   MYSQL_PARSE_ARGUMENTS(ARG
     "LINK_LIBRARIES;DEPENDENCIES;MODULE_OUTPUT_NAME;STATIC_OUTPUT_NAME"
-    "STORAGE_ENGINE;STATIC_ONLY;MODULE_ONLY;MANDATORY;DEFAULT;DISABLED;RECOMPILE_FOR_EMBEDDED"
+    "STORAGE_ENGINE;STATIC_ONLY;MODULE_ONLY;MANDATORY;DEFAULT;DISABLED;RECOMPILE_FOR_EMBEDDED;SKIP_PROBES"
     ${ARGN}
   )
   
@@ -170,7 +170,9 @@
 
     ADD_VERSION_INFO(${target} MODULE SOURCES)
     ADD_LIBRARY(${target} MODULE ${SOURCES}) 
+    IF(NOT ARG_SKIP_PROBES)
     DTRACE_INSTRUMENT(${target})
+    ENDIF()
     SET_TARGET_PROPERTIES (${target} PROPERTIES PREFIX ""
       COMPILE_DEFINITIONS "MYSQL_DYNAMIC_PLUGIN")
     TARGET_LINK_LIBRARIES (${target} mysqlservices)
--- Percona-Server-5.5.32-rel31.0/storage/sphinx/CMakeLists.txt.orig	2012-11-05 12:48:48.000000000 +0100
+++ Percona-Server-5.5.32-rel31.0/storage/sphinx/CMakeLists.txt	2013-07-12 17:57:36.740000378 +0200
@@ -12,5 +12,5 @@
 ADD_LIBRARY(sphinx ha_sphinx.cc snippets_udf.cc)
 ELSE()
 SET(SPHINX_PLUGIN_DYNAMIC "ha_sphinx")
-MYSQL_ADD_PLUGIN(sphinx ${SPHINX_SOURCES} STORAGE_ENGINE MODULE_ONLY LINK_LIBRARIES mysys)
+MYSQL_ADD_PLUGIN(sphinx ${SPHINX_SOURCES} STORAGE_ENGINE MODULE_ONLY SKIP_PROBES LINK_LIBRARIES mysys)
 ENDIF()
--- Percona-Server-5.5.32-rel31.0/plugin/percona-pam-for-mysql/CMakeLists.txt.orig	2013-07-01 05:16:34.000000000 +0200
+++ Percona-Server-5.5.32-rel31.0/plugin/percona-pam-for-mysql/CMakeLists.txt	2013-07-12 18:28:41.576588784 +0200
@@ -31,6 +31,6 @@
   SET(AUTH_PAM_COMPAT_SOURCES ${AUTH_PAM_COMMON_SOURCES} src/auth_pam_compat.c)
   MYSQL_ADD_PLUGIN(auth_pam ${AUTH_PAM_SOURCES} LINK_LIBRARIES pam MODULE_ONLY)
   MYSQL_ADD_PLUGIN(auth_pam_compat ${AUTH_PAM_COMPAT_SOURCES} LINK_LIBRARIES pam MODULE_ONLY)
-  MYSQL_ADD_PLUGIN(dialog src/dialog.c LINK_LIBRARIES mysqlclient MODULE_ONLY)
+  MYSQL_ADD_PLUGIN(dialog src/dialog.c LINK_LIBRARIES mysqlclient MODULE_ONLY SKIP_PROBES)
 ENDIF(HAVE_PAM AND HAVE_GETPWNAM_R AND HAVE_GETGRGID_R)
 ENDIF(WITH_PAM)
